File: backend/eBooking/Persistence/Models/PriceAdjustment.cs
namespace Persistence.Models
{
    public class PriceAdjustment : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public decimal PercentageModifier { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsCumulative { get; set; }

        /// <summary>Admin/employee koji je kreirao pravilo (audit). Server-side se popunjava iz JWT-a.</summary>
        public int? CreatedByUserId { get; set; }
        public User? CreatedByUser { get; set; }

        /// <summary>Null = važi za sve hotele (sajt-wide). Popunjeno = važi samo za taj hotel.</summary>
        public int? HotelId { get; set; }
        public Hotel? Hotel { get; set; }
    }
}

File: Mobile/hotel_booking_app/lib/models/price_adjustment.dart
class PriceAdjustment {
  final int id;
  final String name;
  final double percentageModifier;
  final DateTime startDate;
  final DateTime endDate;
  final bool isCumulative;

  PriceAdjustment({
    required this.id,
    required this.name,
    required this.percentageModifier,
    required this.startDate,
    required this.endDate,
    required this.isCumulative,
  });

  factory PriceAdjustment.fromJson(Map<String, dynamic> json) => PriceAdjustment(
        id: json['id'] ?? 0,
        name: json['name'] ?? '',
        percentageModifier: (json['percentageModifier'] ?? 0).toDouble(),
        startDate: DateTime.tryParse(json['startDate'] ?? '') ?? DateTime.now(),
        endDate: DateTime.tryParse(json['endDate'] ?? '') ?? DateTime.now(),
        isCumulative: json['isCumulative'] ?? false,
      );
}
